pkg/csi/cinder/openstack/openstack_volumes.go

```diff
@@ -241,7 +191,7 @@ func (os *OpenStack) DetachVolume(instanceID, volumeID string) error {
 		return fmt.Errorf("can not detach volume %s, its status is %s", volume.Name, volume.Status)
 	}
 
-	if volume.AttachedServerId != instanceID {
+	if volume.Attachments[0].ServerID != instanceID {
```

Comment:
need to add a check here before len(volume.Attachments) > 0 , else the call would fail?